FirebaseMessaging/Sources/Token/FIRMessagingAPNSInfo.m

```diff
@@ -68,14 +68,17 @@ - (id)copyWithZone:(NSZone *)zone {
   return clone;
 }
 
-#pragma mark - NSCoding
-
+#pragma mark - NSSecureCoding
 + (BOOL)supportsSecureCoding {
   return YES;
 }
 
+- (BOOL)requiresSecureCoding {
+  return YES;
+}
+
 - (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
-  id deviceToken = [aDecoder decodeObjectForKey:kFIRInstanceIDAPNSInfoTokenKey];
+  id deviceToken = [aDecoder decodeObjectOfClass:NSData.self forKey:kFIRInstanceIDAPNSInfoTokenKey];
```

Comment:
Just and observation and I cannot remember since it's been a while since I really messed with Objc, those "id" types could be changed as well? 

`NSData *deviceToken = [aDecoder decodeObjectOfClass:NSData.self forKey:kFIRInstanceIDAPNSInfoTokenKey];`

Reply:
While that is true, I tried to keep changes to a minimum while getting rid of [ decodeObjectForKey:].  The id type itself is not actually a risk.